hyperspy/_components/expression.py

```diff
@@ -118,6 +118,11 @@ class docstring.
             self.__doc__ = _CLASS_DOC % (
                 name, sympy.latex(_parse_substitutions(expression)))
 
+        for parameter in self._parameter_strings:
```

Comment:
why such a convoluted way? why not just
```python
for par in self.parameters:
    par.is_linear = False
```
at this point the component is close to just normal ones and populated.